src/content/datachannel/filetransfer/js/main.js

```diff
@@ -256,24 +256,27 @@ function displayStats() {
     if (adapter.browserDetails.browser === 'chrome') {
       // TODO: once https://code.google.com/p/webrtc/issues/detail?id=4321
       // lands those stats should be preferrred over the connection stats.
-      remoteConnection.getStats(null, function(stats) {
-        for (var key in stats) {
-          var res = stats[key];
-          if (timestampPrev === res.timestamp) {
+      remoteConnection.getStats().then(function(stats) {
```

Comment:
It looks like this PR is doing a lot of different things, can you split it up into multiple PRs for easier reviewability (for example one PR per sample)? If it has to be reverted it's not an all-or-nothing thing and can more easily be re-landed too.